pkg/controller/association/reconciler_test.go

```diff
@@ -1139,3 +1160,79 @@ func equalKeys(t *testing.T, a map[string][]byte, b map[string][]byte) {
 		require.True(t, found, "key %s not found", key)
 	}
 }
+
+func TestReconciler_ReconcileSecretRef(t *testing.T) {
+	// Kibana references ES with a custom secret, but neither the secret nor association conf exist yet
+	kb := sampleKibanaNoEsRef()
+	kb.Spec = kbv1.KibanaSpec{ElasticsearchRef: commonv1.ObjectSelector{Name: "sample-es-ref-secret", Namespace: "kbname", Type: commonv1.ObjectTypeSecret}}
+
+	require.Empty(t, kb.Annotations[kb.EsAssociation().AssociationConfAnnotationName()])
+	r := testReconciler(&kb, &sampleES, &esHTTPPublicCertsSecret)
+	// no resources are watched yet
+	require.Empty(t, r.watches.Secrets.Registrations())
+	require.Empty(t, r.watches.ReferencedResources.Registrations())
+	// run the reconciliation
+	results, err := r.Reconcile(context.Background(), reconcile.Request{NamespacedName: k8s.ExtractNamespacedName(&kb)})
+	// expect and error due to the missing secret
```

Comment:
Yes, I meant "expect an error due to the missing secret".
This is something for which I hesitate because currently:
- we accept that an associated resource does not exist (yet), set status to Pending and remove the existing association conf, without returning error
- we don't accept that a custom service does not exist if one is used, we stop to reconcile with an error